File: src/integration/workflows.ts
/**
 * Workflow Manager - Predefined Workflow Patterns
 * 
 * Provides predefined workflow patterns for common reading scenarios.
 */

import { EngineOrchestrator, EngineConfig, ComprehensiveReading } from './orchestrator';
import { ResultSynthesizer, SynthesisResult } from './synthesizer';
import type { EngineName } from '../types/engines';
import { BaseEngineInput } from '../engines/core/types';

export interface WorkflowDefinition {
  name: string;
  description: string;
  engines: EngineName[];
  executionMode: 'parallel' | 'sequential';
  options: WorkflowOptions;
}

export interface WorkflowOptions {
  includeDivination?: boolean;
  analysisDepth?: 'basic' | 'standard' | 'deep';
  format?: 'standard' | 'mystical' | 'witnessOS';
  focusArea?: string;
}

export interface WorkflowResult {
  workflowName: string;
  timestamp: string;
  inputData: Record<string, any>;
  engineResults: Record<string, any>;
  synthesis: SynthesisResult;
  recommendations: string[];
  nextSteps: string[];
  processingTime: number;
}

export class WorkflowManager {
  private orchestrator: EngineOrchestrator;
  private synthesizer: ResultSynthesizer;
  private workflows: Map<string, WorkflowDefinition>;

  constructor() {
    this.orchestrator = new EngineOrchestrator();
    this.synthesizer = new ResultSynthesizer();
    this.workflows = new Map();
    this.initializeWorkflows();
  }

  /**
   * Initialize predefined workflows
   */
  private initializeWorkflows(): void {
    // Complete Natal Reading
    this.workflows.set('complete_natal', {
      name: 'Complete Natal Analysis',
      description: 'Comprehensive natal chart analysis using all available engines',
      engines: ['numerology', 'human_design', 'vimshottari', 'gene_keys', 'biorhythm', 'enneagram', 'sacred_geometry'],
      executionMode: 'parallel',
      options: {
        includeDivination: false,
        analysisDepth: 'deep',
        format: 'witnessOS'
      }
    });

    // Relationship Compatibility
    this.workflows.set('relationship_compatibility', {
      name: 'Relationship Compatibility Analysis',
      description: 'Two-person compatibility analysis across multiple systems',
      engines: ['numerology', 'human_design', 'enneagram'],
      executionMode: 'parallel',
      options: {
        includeDivination: false,
        analysisDepth: 'standard',
        format: 'standard',
        focusArea: 'relationships'
      }
    });

    // Career Guidance
    this.workflows.set('career_guidance', {
      name: 'Career and Life Purpose Guidance',
      description: 'Career and life purpose guidance using purpose-oriented engines',
      engines: ['numerology', 'human_design', 'gene_keys', 'enneagram'],
      executionMode: 'sequential',
      options: {
        includeDivination: true,
        analysisDepth: 'deep',
        format: 'witnessOS',
        focusArea: 'career'
      }
    });

    // Spiritual Development
    this.workflows.set('spiritual_development', {
      name: 'Spiritual Development Path',
      description: 'Consciousness evolution and spiritual development guidance',
      engines: ['gene_keys', 'sacred_geometry', 'vimshottari', 'iching', 'enneagram'],
      executionMode: 'sequential',
      options: {
        includeDivination: true,
        analysisDepth: 'deep',
        format: 'mystical',
        focusArea: 'spirituality'
      }
    });

    // Life Transition Support
    this.workflows.set('life_transition', {
      name: 'Life Transition Support',
      description: 'Support for major life transitions and changes',
      engines: ['biorhythm', 'vimshottari', 'tarot', 'iching', 'enneagram'],
      executionMode: 'sequential',
      options: {
        includeDivination: true,
        analysisDepth: 'standard',
        format: 'witnessOS',
        focusArea: 'transition'
      }
    });

    // Daily Guidance
    this.workflows.set('daily_guidance', {
      name: 'Daily Energy Optimization',
      description: 'Daily energy optimization and guidance',
      engines: ['biorhythm', 'numerology', 'tarot'],
      executionMode: 'parallel',
      options: {
        includeDivination: true,
        analysisDepth: 'basic',
        format: 'standard'
      }
    });

    // Shadow Work
    this.workflows.set('shadow_work', {
      name: 'Shadow Integration Work',
      description: 'Shadow integration and healing guidance',
      engines: ['enneagram', 'gene_keys', 'tarot', 'human_design'],
      executionMode: 'sequential',
      options: {
        includeDivination: true,
        analysisDepth: 'deep',
        format: 'mystical',
        focusArea: 'shadow'
      }
    });

    // Manifestation Timing
    this.workflows.set('manifestation_timing', {
      name: 'Manifestation Timing Optimization',
      description: 'Optimal timing for manifestation and goal achievement',
      engines: ['vimshottari', 'biorhythm', 'numerology', 'sigil_forge'],
      executionMode: 'sequential',
      options: {
        includeDivination: false,
        analysisDepth: 'standard',
        format: 'witnessOS',
        focusArea: 'manifestation'
      }
    });
  }

  /**
   * Get list of available workflows
   */
  getAvailableWorkflows(): WorkflowDefinition[] {
    return Array.from(this.workflows.values());
  }

  /**
   * Get workflow definition by name
   */
  getWorkflow(workflowName: string): WorkflowDefinition | undefined {
    return this.workflows.get(workflowName);
  }

  /**
   * Run a predefined workflow
   */
  async runWorkflow(
    workflowName: string,
    inputData: Record<string, any>,
    customOptions?: Partial<WorkflowOptions>
  ): Promise<WorkflowResult> {
    const startTime = Date.now();
    
    const workflow = this.workflows.get(workflowName);
    if (!workflow) {
      throw new Error(`Unknown workflow: ${workflowName}`);
    }

    // Merge custom options with workflow defaults
    const options = { ...workflow.options, ...customOptions };

    // Prepare engine configurations
    const engineConfigs: EngineConfig[] = workflow.engines.map(engineName => ({
      name: engineName,
      input: this.prepareWorkflowInput(inputData, engineName, options)
    }));

    // Add divination engines if requested
    if (options.includeDivination && !workflow.engines.includes('tarot')) {
      engineConfigs.push({
        name: 'tarot',
        input: this.prepareDivinationInput(inputData, 'tarot', options)
      });
    }

    // Execute workflow
    let engineResults;
    if (workflow.executionMode === 'parallel') {
      engineResults = await this.orchestrator.runParallelEngines(engineConfigs);
    } else {
      engineResults = await this.orchestrator.runSequentialEngines(engineConfigs);
    }

    // Synthesize results
    const synthesis = this.synthesizer.synthesizeReading(engineResults);

    // Generate workflow-specific recommendations
    const recommendations = this.generateWorkflowRecommendations(workflowName, synthesis, options);
    const nextSteps = this.generateNextSteps(workflowName, synthesis);

    const result: WorkflowResult = {
      workflowName,
      timestamp: new Date().toISOString(),
      inputData,
      engineResults,
      synthesis,
      recommendations,
      nextSteps,
      processingTime: Date.now() - startTime
    };

    return result;
  }

  /**
   * Prepare input for workflow engines
   */
  private prepareWorkflowInput(
    inputData: Record<string, any>,
    engineName: EngineName,
    options: WorkflowOptions
  ): BaseEngineInput {
    const baseInput = {
      birthDate: inputData.birthDate || inputData.date,
      birthTime: inputData.birthTime || inputData.time,
      birthLocation: inputData.birthLocation || inputData.location,
      fullName: inputData.fullName || inputData.name
    };

    // Engine-specific input preparation
    switch (engineName) {
      case 'enneagram':
        return {
          ...baseInput,
          identificationMethod: 'intuitive',
          behavioralDescription: `Person seeking ${options.focusArea || 'general'} guidance`,
          includeWings: options.analysisDepth !== 'basic',
          includeArrows: options.analysisDepth === 'deep',
          includeInstincts: options.analysisDepth === 'deep',
          focusArea: options.focusArea as any
        };

      case 'gene_keys':
        return {
          ...baseInput,
          focusSequence: options.focusArea === 'relationships' ? 'venus' : 
                        options.focusArea === 'career' ? 'pearl' : 'activation'
        };

      case 'sacred_geometry':
        return {
          ...baseInput,
          patternType: 'auto',
          includePersonalization: true,
          intention: `${options.focusArea || 'general'} guidance and support`
        };

      case 'sigil_forge':
        return {
          ...baseInput,
          intention: `Support for ${options.focusArea || 'personal development'}`,
          generationMethod: 'personal',
          includePersonalization: true
        };

      default:
        return baseInput;
    }
  }

  /**
   * Prepare divination input for workflows
   */
  private prepareDivinationInput(
    inputData: Record<string, any>,
    engineName: string,
    options: WorkflowOptions
  ): BaseEngineInput {
    const question = this.generateWorkflowQuestion(inputData, options);

    if (engineName === 'tarot') {
      return {
        question,
        spreadType: options.analysisDepth === 'deep' ? 'celtic_cross' : 'three_card',
        includeReversed: true
      };
    } else if (engineName === 'iching') {
      return {
        question,
        method: 'coins'
      };
    }

    return { question };
  }

  /**
   * Generate workflow-specific question
   */
  private generateWorkflowQuestion(inputData: Record<string, any>, options: WorkflowOptions): string {
    const name = inputData.fullName || inputData.name || 'this person';
    const focus = options.focusArea || 'life path';

    switch (options.focusArea) {
      case 'career':
        return `What guidance can you provide for ${name}'s career path and professional development?`;
      case 'relationships':
        return `What insights can you offer about ${name}'s relationship patterns and connections?`;
      case 'spirituality':
        return `What spiritual guidance and development opportunities are available for ${name}?`;
      case 'shadow':
        return `What shadow aspects and integration opportunities should ${name} be aware of?`;
      case 'transition':
        return `What guidance can you provide for ${name}'s current life transition?`;
      case 'manifestation':
        return `What is the optimal timing and approach for ${name}'s manifestation goals?`;
      default:
        return `What general life guidance and insights are available for ${name} at this time?`;
    }
  }

  /**
   * Generate workflow-specific recommendations
   */
  private generateWorkflowRecommendations(
    workflowName: string,
    synthesis: SynthesisResult,
    options: WorkflowOptions
  ): string[] {
    const baseRecommendations = synthesis.integrationGuidance.map(g => g.guidance);

    // Add workflow-specific recommendations
    const workflowSpecific: string[] = [];

    switch (workflowName) {
      case 'complete_natal':
        workflowSpecific.push(
          'Study your complete consciousness blueprint for deep self-understanding',
          'Focus on integrating insights from all systems for holistic growth',
          'Use this analysis as a foundation for ongoing self-development'
        );
        break;

      case 'career_guidance':
        workflowSpecific.push(
          'Align your career choices with your natural gifts and purpose',
          'Consider how your personality type influences your work style',
          'Look for opportunities that match your energy patterns'
        );
        break;

      case 'spiritual_development':
        workflowSpecific.push(
          'Establish a regular spiritual practice based on your blueprint',
          'Work with the identified growth edges for consciousness expansion',
          'Use sacred geometry and symbols for meditation and focus'
        );
        break;

      case 'shadow_work':
        workflowSpecific.push(
          'Approach shadow work with compassion and patience',
          'Use the identified patterns for conscious integration',
          'Consider working with a qualified guide for deep shadow work'
        );
        break;

      default:
        workflowSpecific.push(
          'Regular review and integration of insights is recommended',
          'Consider follow-up readings to track your evolution'
        );
    }

    return [...baseRecommendations.slice(0, 5), ...workflowSpecific];
  }

  /**
   * Generate next steps for the workflow
   */
  private generateNextSteps(workflowName: string, synthesis: SynthesisResult): string[] {
    const nextSteps: string[] = [];

    // Universal next steps
    nextSteps.push(
      'Reflect on the insights and correlations identified',
      'Choose 2-3 key recommendations to focus on initially'
    );

    // Workflow-specific next steps
    switch (workflowName) {
      case 'complete_natal':
        nextSteps.push(
          'Create a personal development plan based on your blueprint',
          'Consider specialized readings for specific life areas',
          'Schedule follow-up readings to track your evolution'
        );
        break;

      case 'career_guidance':
        nextSteps.push(
          'Research career paths that align with your identified gifts',
          'Network with people in fields that match your energy type',
          'Consider additional training or education in aligned areas'
        );
        break;

      case 'spiritual_development':
        nextSteps.push(
          'Begin or deepen your meditation practice',
          'Study the spiritual traditions that resonate with your blueprint',
          'Connect with like-minded spiritual communities'
        );
        break;

      case 'life_transition':
        nextSteps.push(
          'Use the timing insights to plan your transition phases',
          'Prepare for the challenges and opportunities identified',
          'Seek support during transition periods'
        );
        break;

      default:
        nextSteps.push(
          'Apply the insights to your daily life gradually',
          'Monitor your progress and adjust as needed'
        );
    }

    return nextSteps;
  }

  /**
   * Create custom workflow
   */
  createCustomWorkflow(
    name: string,
    engines: EngineName[],
    options: WorkflowOptions,
    executionMode: 'parallel' | 'sequential' = 'parallel'
  ): void {
    this.workflows.set(name, {
      name,
      description: `Custom workflow: ${name}`,
      engines,
      executionMode,
      options
    });
  }
}

export const workflowManager = new WorkflowManager();
export default workflowManager;

File: docs/api/engines/integration/workflows.py
"""
Workflow Manager - Predefined Multi-Engine Workflows

Provides common workflow patterns for different types of readings
and consciousness analysis scenarios.
"""

from typing import Dict, List, Any, Optional
from datetime import datetime, date
import logging

try:
    from .orchestrator import EngineOrchestrator
    from .synthesis import ResultSynthesizer
except ImportError:
    import sys
    import os
    sys.path.insert(0, os.path.dirname(os.path.abspath(__file__)))
    from orchestrator import EngineOrchestrator
    from synthesis import ResultSynthesizer


class WorkflowManager:
    """
    Manages predefined workflows for common reading scenarios
    """
    
    def __init__(self):
        self.orchestrator = EngineOrchestrator()
        self.synthesizer = ResultSynthesizer()
        self.logger = logging.getLogger(__name__)
        
        # Define workflow templates
        self.workflows = {
            'complete_natal': self._complete_natal_workflow,
            'relationship_compatibility': self._relationship_compatibility_workflow,
            'career_guidance': self._career_guidance_workflow,
            'spiritual_development': self._spiritual_development_workflow,
            'life_transition': self._life_transition_workflow,
            'daily_guidance': self._daily_guidance_workflow,
            'shadow_work': self._shadow_work_workflow,
            'manifestation_timing': self._manifestation_timing_workflow
        }
    
    def run_workflow(self, workflow_name: str, input_data: Dict, 
                    options: Optional[Dict] = None) -> Dict[str, Any]:
        """
        Run a predefined workflow
        
        Args:
            workflow_name: Name of the workflow to run
            input_data: Input data for the workflow
            options: Optional workflow customization options
        """
        if workflow_name not in self.workflows:
            raise ValueError(f"Unknown workflow: {workflow_name}")
        
        self.logger.info(f"Starting workflow: {workflow_name}")
        
        # Run the workflow
        workflow_func = self.workflows[workflow_name]
        results = workflow_func(input_data, options or {})
        
        # Synthesize results
        synthesis = self.synthesizer.synthesize_reading(results['engine_results'])
        
        # Combine workflow results with synthesis
        final_result = {
            'workflow_name': workflow_name,
            'timestamp': datetime.now().isoformat(),
            'input_data': input_data,
            'options': options,
            'engine_results': results['engine_results'],
            'synthesis': synthesis,
            'workflow_insights': results.get('workflow_insights', {}),
            'recommendations': results.get('recommendations', [])
        }
        
        self.logger.info(f"Completed workflow: {workflow_name}")
        return final_result
    
    def _complete_natal_workflow(self, input_data: Dict, options: Dict) -> Dict[str, Any]:
        """
        Complete natal chart analysis using all available engines
        """
        birth_data = input_data
        
        # Define engines for complete natal reading
        engines = ['numerology', 'biorhythm', 'human_design', 'vimshottari', 'gene_keys']
        
        # Add divination engines if requested
        if options.get('include_divination', True):
            engines.extend(['tarot', 'iching'])
        
        # Run engines
        engine_results = self.orchestrator.create_comprehensive_reading(birth_data, engines)
        
        # Workflow-specific insights
        workflow_insights = {
            'natal_themes': self._extract_natal_themes(engine_results['results']),
            'life_purpose_synthesis': self._synthesize_life_purpose(engine_results['results']),
            'personality_integration': self._analyze_personality_integration(engine_results['results'])
        }
        
        recommendations = self._generate_natal_recommendations(engine_results['results'])
        
        return {
            'engine_results': engine_results['results'],
            'workflow_insights': workflow_insights,
            'recommendations': recommendations
        }
    
    def _relationship_compatibility_workflow(self, input_data: Dict, options: Dict) -> Dict[str, Any]:
        """
        Relationship compatibility analysis for two people
        """
        person1_data = input_data['person1']
        person2_data = input_data['person2']
        
        # Run engines for both people
        person1_results = self.orchestrator.create_comprehensive_reading(
            person1_data, ['numerology', 'biorhythm', 'human_design', 'gene_keys']
        )
        person2_results = self.orchestrator.create_comprehensive_reading(
            person2_data, ['numerology', 'biorhythm', 'human_design', 'gene_keys']
        )
        
        # Compatibility analysis
        compatibility = self._analyze_compatibility(
            person1_results['results'], 
            person2_results['results']
        )
        
        workflow_insights = {
            'compatibility_score': compatibility['overall_score'],
            'strengths': compatibility['strengths'],
            'challenges': compatibility['challenges'],
            'growth_opportunities': compatibility['growth_opportunities']
        }
        
        recommendations = self._generate_relationship_recommendations(compatibility)
        
        # Combine results
        combined_results = {
            'person1': person1_results['results'],
            'person2': person2_results['results'],
            'compatibility': compatibility
        }
        
        return {
            'engine_results': combined_results,
            'workflow_insights': workflow_insights,
            'recommendations': recommendations
        }
    
    def _career_guidance_workflow(self, input_data: Dict, options: Dict) -> Dict[str, Any]:
        """
        Career and life purpose guidance workflow
        """
        birth_data = input_data
        
        # Focus on engines that provide career insights
        engines = ['numerology', 'human_design', 'gene_keys', 'vimshottari']
        
        engine_results = self.orchestrator.create_comprehensive_reading(birth_data, engines)
        
        # Career-specific analysis
        career_analysis = self._analyze_career_potential(engine_results['results'])
        
        workflow_insights = {
            'career_themes': career_analysis['themes'],
            'natural_talents': career_analysis['talents'],
            'optimal_work_environment': career_analysis['environment'],
            'timing_considerations': career_analysis['timing']
        }
        
        recommendations = self._generate_career_recommendations(career_analysis)
        
        return {
            'engine_results': engine_results['results'],
            'workflow_insights': workflow_insights,
            'recommendations': recommendations
        }
    
    def _spiritual_development_workflow(self, input_data: Dict, options: Dict) -> Dict[str, Any]:
        """
        Spiritual development and consciousness evolution workflow
        """
        birth_data = input_data
        
        # Focus on spiritual/consciousness engines
        engines = ['gene_keys', 'human_design', 'iching', 'vimshottari']
        
        # Add sacred geometry if available
        if 'sacred_geometry' in self.orchestrator.get_available_engines():
            engines.append('sacred_geometry')
        
        engine_results = self.orchestrator.create_comprehensive_reading(birth_data, engines)
        
        # Spiritual development analysis
        spiritual_analysis = self._analyze_spiritual_path(engine_results['results'])
        
        workflow_insights = {
            'current_evolutionary_stage': spiritual_analysis['stage'],
            'shadow_work_areas': spiritual_analysis['shadow_areas'],
            'gift_activation_potential': spiritual_analysis['gifts'],
            'spiritual_practices': spiritual_analysis['practices']
        }
        
        recommendations = self._generate_spiritual_recommendations(spiritual_analysis)
        
        return {
            'engine_results': engine_results['results'],
            'workflow_insights': workflow_insights,
            'recommendations': recommendations
        }
    
    def _life_transition_workflow(self, input_data: Dict, options: Dict) -> Dict[str, Any]:
        """
        Life transition guidance workflow
        """
        birth_data = input_data
        transition_type = options.get('transition_type', 'general')
        
        # Focus on timing and transition engines
        engines = ['biorhythm', 'vimshottari', 'tarot', 'iching']
        
        engine_results = self.orchestrator.create_comprehensive_reading(birth_data, engines)
        
        # Transition-specific analysis
        transition_analysis = self._analyze_transition_timing(
            engine_results['results'], 
            transition_type
        )
        
        workflow_insights = {
            'transition_timing': transition_analysis['timing'],
            'supportive_energies': transition_analysis['support'],
            'potential_challenges': transition_analysis['challenges'],
            'optimal_strategies': transition_analysis['strategies']
        }
        
        recommendations = self._generate_transition_recommendations(transition_analysis)
        
        return {
            'engine_results': engine_results['results'],
            'workflow_insights': workflow_insights,
            'recommendations': recommendations
        }
    
    def _daily_guidance_workflow(self, input_data: Dict, options: Dict) -> Dict[str, Any]:
        """
        Daily guidance and energy optimization workflow
        """
        birth_data = input_data
        target_date = options.get('target_date', date.today())
        
        # Focus on daily/cyclical engines
        engines = ['biorhythm', 'numerology']
        
        # Add divination for daily guidance
        if options.get('include_divination', True):
            engines.extend(['tarot', 'iching'])
        
        engine_results = self.orchestrator.create_comprehensive_reading(birth_data, engines)
        
        # Daily guidance analysis
        daily_analysis = self._analyze_daily_energies(engine_results['results'], target_date)
        
        workflow_insights = {
            'energy_forecast': daily_analysis['forecast'],
            'optimal_activities': daily_analysis['activities'],
            'timing_recommendations': daily_analysis['timing'],
            'awareness_points': daily_analysis['awareness']
        }
        
        recommendations = self._generate_daily_recommendations(daily_analysis)
        
        return {
            'engine_results': engine_results['results'],
            'workflow_insights': workflow_insights,
            'recommendations': recommendations
        }
    
    def _shadow_work_workflow(self, input_data: Dict, options: Dict) -> Dict[str, Any]:
        """
        Shadow work and integration workflow
        """
        birth_data = input_data
        
        # Focus on engines that reveal shadow aspects
        engines = ['gene_keys', 'human_design', 'enneagram']
        
        # Add divination for shadow exploration
        engines.extend(['tarot', 'iching'])
        
        engine_results = self.orchestrator.create_comprehensive_reading(birth_data, engines)
        
        # Shadow work analysis
        shadow_analysis = self._analyze_shadow_patterns(engine_results['results'])
        
        workflow_insights = {
            'shadow_themes': shadow_analysis['themes'],
            'integration_opportunities': shadow_analysis['integration'],
            'gift_potential': shadow_analysis['gifts'],
            'healing_pathways': shadow_analysis['healing']
        }
        
        recommendations = self._generate_shadow_work_recommendations(shadow_analysis)
        
        return {
            'engine_results': engine_results['results'],
            'workflow_insights': workflow_insights,
            'recommendations': recommendations
        }
    
    def _manifestation_timing_workflow(self, input_data: Dict, options: Dict) -> Dict[str, Any]:
        """
        Manifestation timing and energy alignment workflow
        """
        birth_data = input_data
        intention = options.get('intention', '')
        
        # Focus on timing and energy engines
        engines = ['biorhythm', 'vimshottari', 'numerology']
        
        # Add sacred geometry for manifestation support
        if 'sacred_geometry' in self.orchestrator.get_available_engines():
            engines.append('sacred_geometry')
        
        engine_results = self.orchestrator.create_comprehensive_reading(birth_data, engines)
        
        # Manifestation timing analysis
        timing_analysis = self._analyze_manifestation_timing(
            engine_results['results'], 
            intention
        )
        
        workflow_insights = {
            'optimal_timing': timing_analysis['timing'],
            'energy_alignment': timing_analysis['alignment'],
            'supportive_practices': timing_analysis['practices'],
            'manifestation_strategy': timing_analysis['strategy']
        }
        
        recommendations = self._generate_manifestation_recommendations(timing_analysis)
        
        return {
            'engine_results': engine_results['results'],
            'workflow_insights': workflow_insights,
            'recommendations': recommendations
        }
    
    def get_available_workflows(self) -> List[str]:
        """Get list of available workflows"""
        return list(self.workflows.keys())
    
    def get_workflow_description(self, workflow_name: str) -> str:
        """Get description of a specific workflow"""
        descriptions = {
            'complete_natal': 'Comprehensive natal chart analysis using all engines',
            'relationship_compatibility': 'Two-person compatibility analysis',
            'career_guidance': 'Career and life purpose guidance',
            'spiritual_development': 'Spiritual evolution and consciousness development',
            'life_transition': 'Guidance for major life transitions',
            'daily_guidance': 'Daily energy optimization and guidance',
            'shadow_work': 'Shadow integration and healing work',
            'manifestation_timing': 'Optimal timing for manifestation and goal achievement'
        }
        return descriptions.get(workflow_name, 'No description available')
    
    # Simplified helper methods (would be fully implemented)
    def _extract_natal_themes(self, results: Dict) -> List[str]:
        return ['Life purpose', 'Personality integration', 'Karmic patterns']
    
    def _synthesize_life_purpose(self, results: Dict) -> str:
        return 'Integrated life purpose synthesis'
    
    def _analyze_personality_integration(self, results: Dict) -> Dict:
        return {'integration_level': 'High', 'areas_for_growth': []}
    
    def _generate_natal_recommendations(self, results: Dict) -> List[str]:
        return ['Focus on personal development', 'Explore creative expression']
    
    def _analyze_compatibility(self, person1: Dict, person2: Dict) -> Dict:
        return {
            'overall_score': 0.75,
            'strengths': ['Communication', 'Shared values'],
            'challenges': ['Different life rhythms'],
            'growth_opportunities': ['Mutual learning']
        }
    
    def _generate_relationship_recommendations(self, compatibility: Dict) -> List[str]:
        return ['Practice active listening', 'Honor different rhythms']
    
    def _analyze_career_potential(self, results: Dict) -> Dict:
        return {
            'themes': ['Leadership', 'Communication'],
            'talents': ['Strategic thinking', 'Empathy'],
            'environment': 'Collaborative team setting',
            'timing': 'Current period favorable for career growth'
        }
    
    def _generate_career_recommendations(self, analysis: Dict) -> List[str]:
        return ['Develop leadership skills', 'Seek mentorship opportunities']
    
    def _analyze_spiritual_path(self, results: Dict) -> Dict:
        return {
            'stage': 'Integration phase',
            'shadow_areas': ['Control patterns'],
            'gifts': ['Intuitive wisdom'],
            'practices': ['Meditation', 'Shadow work']
        }
    
    def _generate_spiritual_recommendations(self, analysis: Dict) -> List[str]:
        return ['Daily meditation practice', 'Work with shadow aspects']
    
    def _analyze_transition_timing(self, results: Dict, transition_type: str) -> Dict:
        return {
            'timing': 'Favorable for next 3 months',
            'support': ['Strong intuitive guidance'],
            'challenges': ['Resistance to change'],
            'strategies': ['Gradual implementation']
        }
    
    def _generate_transition_recommendations(self, analysis: Dict) -> List[str]:
        return ['Take gradual steps', 'Trust intuitive guidance']
    
    def _analyze_daily_energies(self, results: Dict, target_date: date) -> Dict:
        return {
            'forecast': 'High creative energy',
            'activities': ['Creative projects', 'Communication'],
            'timing': 'Morning hours most productive',
            'awareness': ['Avoid overcommitment']
        }
    
    def _generate_daily_recommendations(self, analysis: Dict) -> List[str]:
        return ['Focus on creative work in morning', 'Schedule rest periods']
    
    def _analyze_shadow_patterns(self, results: Dict) -> Dict:
        return {
            'themes': ['Control', 'Perfectionism'],
            'integration': ['Self-compassion practices'],
            'gifts': ['Attention to detail', 'High standards'],
            'healing': ['Inner child work']
        }
    
    def _generate_shadow_work_recommendations(self, analysis: Dict) -> List[str]:
        return ['Practice self-compassion', 'Explore inner child healing']
    
    def _analyze_manifestation_timing(self, results: Dict, intention: str) -> Dict:
        return {
            'timing': 'Next new moon cycle optimal',
            'alignment': 'Strong energy alignment present',
            'practices': ['Visualization', 'Gratitude'],
            'strategy': 'Focus on feeling states'
        }
    
    def _generate_manifestation_recommendations(self, analysis: Dict) -> List[str]:
        return ['Begin manifestation work at new moon', 'Focus on feeling states']
